rust/lance/src/index.rs

```diff
@@ -147,6 +146,32 @@ pub trait DatasetIndexExt {
 
     /// Optimize indices.
     async fn optimize_indices(&mut self) -> Result<()>;
+
+    async fn open_generic_index(&self, column: &str, uuid: &str) -> Result<Arc<dyn Index>>;
```

Comment:
The good thing is that our rust API has no stability guarantees. 

But i do wonder `open index` should be a lance-internal API, as users facing API is just doing "search".

Reply:
That's a good idea.  Should I make a `DatasetIndexInternalExt`?  Or just let them be free functions?